xblock/mixins.py

```diff
@@ -516,6 +516,10 @@ def _add_field(self, node, field_name, field):
         as an xml attribute or creates a separate child node.
         """
         value = field.to_string(field.read_from(self))
+
+        if not isinstance(value, basestring):
```

Comment:
On second thought, it might make more sense to change `DateTime` field's `to_json` method to return an empty string instead of `None` when the value is `None`? `DateTime`'s `from_json` already converts empty strings to `None`.